### ISSUE TITLE
[Preview 5] Constant FrameMessages being sent in previewer

### ISSUE THREAD

**Issue by amwx** [AvaloniaUI/Avalonia#10203]
**Describe the bug**
In Preview 5, when a xaml file is open in the designer the previewer process is sending constant FrameMessages for no reason (VS is just sitting open while I type this and we're already at sequence id 17 000+). 

I've traced this back to nightly CI build `11.0.999-cibuild0029216-beta` which was built on Jan 29. I think that corresponds to #10120 as the package is the second nightly package on the 29th and that was the second PR merged, and the PR changes show some changes with the previewer.

![image](https://user-images.githubusercontent.com/40413319/216730737-2ecca5f7-2bfa-4bc9-8517-6ab249115396.png)

**Comment by amwx** [comment]
I dug into this more with the azure logs and found it was actually the PR that removed the legacy renderers (#10078) that is causing this. Seems the compositor doesn't go to sleep and sends constant updates, at least that's what I think is happening. 

cc @kekekeks

**Comment by kuiperzone** [comment]
Yes, we are seeing this here: https://github.com/kuiperzone/AvantGarde/issues/11

We notice that CPU load actually drops when there is continuous mouse movement over the preview window, and increases when movement stops. Suggests a tight loop in the preview host.

**Comment by amwx** [comment]
If this doesn't have a quick solution that can make it into preview6, can the legacy renderer be returned just for the previewer in the meantime? Besides the CPU, the constantly running process is interfering with VS (each frame has 2 bitmaps that are returned, one transparent and one actual - #7639), which seems to be messing with the dispatcher and causes other issues that go away as soon as all the previewer tabs are closed.

**Comment by grokys** [comment]
Seems to be caused by this line:

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Rendering/Composition/Compositor.cs#L74

When running in the previewer this line is called on the UI thread. However the continuation is called in a background thread, resulting in the commit notification being dispatched to the main thread, meaning it arrives too late.

**Comment by grokys** [comment]
Seems to be caused because though the compositor is being run on the main thread, the commit message is posted:

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Rendering/Composition/Compositor.cs#L74

This causes it to arrive after `RemoteServerTopLevelImpl.RenderIfNeeded` exits.

This means that when [`RemoteServerTopLevelImpl.QueueNextRender`](https://github.com/AvaloniaUI/Avalonia/blob/7158628805fac10de6eae75b69f5e8b0e7647440/src/Avalonia.Controls/Remote/Server/RemoteServerTopLevelImpl.cs#L351) is called, `_inRender` has been set to false, causing the top level impl to think that a new render has completed due to an invalidation.

**Comment by Nimred** [comment]
I also noticed this CPU usage issue today in preview6. Just wanted to point out that this isn't only when a designer is being displayed, it also happens with tabs that aren't active. Each open tab with an xaml file and a designer will cause an extra .NET Host process to appear and drain resources, and they remain even when you switch to a C# file tab.

**Comment by grabthar** [comment]
Also affect preview 7
